### PR TITLE
Feature/sequence io

### PR DESCRIPTION
**Pull request by h-2**
very much WIP, but record based reading already works.

Please do not review yet in detail. This PR is just for reference and discussion of the general design.

### CODE REVIEW

**Comment by rrahn** [comment]
!rebuild doc

**Comment by rrahn** [comment]
!doc

**Comment by rrahn** [comment]
!doc

**Comment by rrahn** [comment]
!doc

**Comment by rrahn** [comment]
!doc

**Comment by rrahn** [comment]
!doc

**Comment by rrahn** [comment]
!doc

**Comment by rrahn** [comment]
!doc

**Comment by h-2** [comment]
@rrahn whats going on here? :grin:

**Comment by marehr** [comment]
@h-2 go with the flow

!doc

**Comment by rrahn** [comment]
!doc

File: include/seqan3/io/sequence/sequence_file_traits.hpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+namespace seqan3
+{
+
+
+template <typename t>
+concept bool sequence_file_traits_concept = requires (t v)
+{
+    typename t::format_type;
+    requires detail::all_holdees_meets_sequence_file_format_concept<typename t::format_type>();
```
Comment (smehringer) [inline include/seqan3/io/sequence/sequence_file_traits.hpp]: `meet~~s~~`

holdees :heart:

**Pull request closed without merging**